You-Can-Code-It/fndr
# Edit company

# What this pull request does

## What

This is an Edit feature. It allows the user to change company details so it stays up to date.
This PR is made based by [issue](https://github.com/orgs/You-Can-Code-It/projects/1/views/1?pane=issue&itemId=36983383)

![EDIT](https://github.com/luizfiorentino/fndr/assets/51381521/5020f8a4-4350-4767-8abf-ff078e51f101)

## Why

User can change company details to keep the data about the company is up to date.

## How

- Create an `EditCompanyForm` component based on `AddCompanyForm`
- Style form
- Create `PATCH` method for updating data in database
- Moved edit form from `Modal` to `Edit page`

## Steppings to test this

- `git pull` changes from `development` branch
- `npm i` all updates
- `npx prisma generate` to get all updates from database
- `npx prisma studio` to check the database
- `npm run dev` and open http://localhost:3000/
- On the main page click on `Edit` button on the card
- Change data if needed
- Click `Save` to apply changes -> you'll see `Details page`
- Click `Back` or `Cancel` to go on main page again

## Anything else

I created `BackIcon` component with `<Image>` inside. But Luiz has the same icon in different place and I didn't change his code. Maybe it can be improvements
